.github/workflows/publish-ghcr.yml

```diff
@@ -0,0 +1,70 @@
+#
+name: Create and publish a Docker image
+
+on:
+  workflow_dispatch:
+  
+  release:
+    types: [published]
+
+# Defines three custom environment variables for the workflow. These are used for the Container registry domain, and a name + version for the Docker image that this workflow builds.
+env:
+  REGISTRY: ghcr.io
+  IMAGE_NAME: ${{ github.repository }}
+  VERSION: ${{ github.sha }}
+
+# There are two joba in this workflow. It's configured to run on the latest available version of Ubuntu.
+jobs:
+  build-and-push-image-normal:
+    runs-on: ubuntu-latest
+    # Sets the permissions granted to the `GH_PAT` for the actions in this job.
+    permissions:
+      contents: read
+      packages: write
+      #
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+      # Uses the `docker/login-action` action to log in to the Container registry registry using the account and password that will publish the packages. Once published, the packages are scoped to the account defined here.
+      - name: Log in to the Container registry
+        uses: docker/login-action@65b78e6e13532edd9afa3aa52ac7964289d1a9c1
```

Comment:
Any specific reason to use `@65b78e6e13532edd9afa3aa52ac7964289d1a9c1` instead of something like `@v3`